pkg/tls/tlsmanager.go

```diff
@@ -107,6 +104,7 @@ func (m *Manager) Get(storeName, configName string) (*tls.Config, error) {
 		domainToCheck := types.CanonicalDomain(clientHello.ServerName)
 
 		if isACMETLS(clientHello) {
+			acmeTLSStore := m.getStore(tlsalpn01.ACMETLS1Protocol)
```

Comment:
two things worry me, related to the fact that there's a mutex guarding m for the whole duration of Get.

1) if you're moving m.getStore into the GetCertificate closure, you're taking it out of the mutex protection, therefore potentially introducing a race.

2) (unrelated to your change) since the very first call to getStore potentially modifies m.stores, it might have been a mistake to use a Read lock at the beginning of Get, and it might be necessary to use a RW lock instead.

Reply:
And in that respect, it is also strange that the access to `m.configs[configName].SniStrict` L122 isn't guarded either.

Reply:
Sorry for that, I did not check the code carefully, so it might be wrong regarding mutexes. Also, you are right, only in case of acme store does it provide a solution. The default store's issue can be solved externally, providing a default certificate, but acme's cannot. At least, I did not find a way for it.

Reply:
Would it be enough to lock the mutex inside the closure function?

Reply:
According to your comments and with my changes, an RLock() may be enough at the beginning of Manager.Get(), and in the  tlsConfig.GetCertificate function an RW Lock should be obtained. Am I right?

Reply:
Adding an RWLock within GetCertificate sounds scary and a recipe for deadlocks at first impression, but maybe. But hold on please, I want to understand first 1) why there needs to be a default certificate in the acme store, 2) whether, just like for the default store, we could simply let the configuration provide the default cert for the acme store, and 3) why we only have an RLock (instead of an RWLock) at the beginning of Get.